src/robotFactory.js

```diff
@@ -1,17 +1,74 @@
 'use strict';
 
-function BaseRobot() {
-  // implement
+function BaseRobot(name, weight, coords, chipVersion) {
+  this.name = name;
+  this.weight = weight;
+  this.coords = coords;
+  this.chipVersion = chipVersion;
 }
 
-function FlyingRobot() {
-  // implement
-}
+BaseRobot.prototype.goForward = function(step = 1) {
+  this.coords.y += step;
+};
+
+BaseRobot.prototype.goBack = function(step = 1) {
+  this.coords.y -= step;
+};
+
+BaseRobot.prototype.goRight = function(step = 1) {
+  this.coords.x += step;
+};
+
+BaseRobot.prototype.goLeft = function(step = 1) {
+  this.coords.x -= step;
+};
+
+BaseRobot.prototype.getInfo = function() {
+  return `Robot: ${this.name}, `
+  + `Chip version: ${this.chipVersion}, `
+  + `Weight: ${this.weight}`;
+};
+
+FlyingRobot.prototype = BaseRobot.prototype;
+
+function FlyingRobot(name, weight, coords, chipVersion) {
+  BaseRobot.call(this, name, weight, coords, chipVersion);
 
-function DeliveryDrone() {
-  // implement
+  this.coords = {
+    ...coords,
+    z: 0,
+  };
```

Comment:
Может я что-то упускаю, но ты же вроде уже создала объект coords вверху через функцию BaseRobot. Достаточно просто записать в этот объект координату z
```suggestion
  this.coords.z = 0;
```